smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelParser.java

```diff
@@ -721,10 +722,19 @@ SourceLocation currentLocation() {
     NumberNode parseNumberNode() {
         SourceLocation location = currentLocation();
         String lexeme = ParserUtils.parseNumber(this);
+
         if (lexeme.contains("e") || lexeme.contains(".")) {
```

Comment:
Wait this should be checking for both lower and upper case e - `ParserUtils.parseNumber` supports both and doesn't do any lowercasing.

Reply:
Fixed in a4dcceb.